quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java

```diff
@@ -0,0 +1,326 @@
+            @Override
+            public boolean verify(String s, SSLSession sslSession) {
+                return true;
+            }
+        };
+    }
+
+    private SSLSocketFactory createInsecureSslSocketFactory() throws IOException {
+        TrustManager[] trustAllCerts = new TrustManager[] {new X509TrustManager() {
+            public void checkClientTrusted(final X509Certificate[] chain, final String authType) {
+            }
+
+            public void checkServerTrusted(final X509Certificate[] chain, final String authType) {
+            }
+
+            public X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+        }};
+
+        SSLContext sslContext;
+        SSLSocketFactory socketFactory;
+
+        try {
+            sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null, trustAllCerts, new SecureRandom());
+            socketFactory = sslContext.getSocketFactory();
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+            throw new IOException("Can't create unsecure trust manager");
+        }
+        return socketFactory;
+    }
+
+    private boolean isRunning() {
+        return keycloak != null && keycloak.isAlive();
+    }
+
+    private void asyncReadOutput() {
+        shutdownOutputExecutor();
+        outputExecutor = Executors.newSingleThreadExecutor();
+        outputExecutor.execute(this::readOutput);
+    }
+
+    private void shutdownOutputExecutor() {
+        if (outputExecutor != null) {
+            outputExecutor.shutdown();
+            try {
+                outputExecutor.awaitTermination(30, TimeUnit.SECONDS);
+            } catch (InterruptedException cause) {
+                throw new RuntimeException("Failed to terminate output executor", cause);
+            } finally {
+                outputExecutor = null;
+            }
+        }
+    }
+
+    private void reset() {
+        outputStream.clear();
+        errorStream.clear();
+        exitCode = -1;
+        keycloak = null;
+        shutdownOutputExecutor();
+    }
+
+    private Path tmpDir = null;
+
+    private Path prepareDistribution() {
+        try {
+            tmpDir = Files.createTempDirectory("keycloak-x-it");
+            tmpDir.toFile().deleteOnExit();
+            Path distRootPath = tmpDir.resolve("kc-tests");
+            File distFile = new File("../../../distribution/server-x-dist/target/keycloak.x-" + Version.VERSION_KEYCLOAK + ".zip");
+            if (!distFile.exists()) {
+                throw new RuntimeException("Distribution archive " + distFile.getAbsolutePath() +" doesn't exists");
+            }
+            distRootPath.toFile().mkdirs();
+            String distDirName = distFile.getName().replace("keycloak-server-x-dist", "keycloak.x");
+            Path distPath = distRootPath.resolve(distDirName.substring(0, distDirName.lastIndexOf('.')));
+
+            if (distPath.toFile().exists()) {
```

Comment:
We can not assume tests always want to re-create the dist. There are quite different test scenarios that should require:

* Never reinstall
* Reinstall before all
* Reinstall before each test

A good example is a test the has different steps (test methods) for building and then starting the server using the persited config.

Can we keep the way it was before?